### PR TITLE
Metric design

### PR DESCRIPTION
**Pull request by delhomer**
This PR aims at introducing new metrics to follow the training performance. We design Intersection-over-Union (IoU) as in https://en.wikipedia.org/wiki/Jaccard_index, and the Dice coefficient, as in https://en.wikipedia.org/wiki/S%C3%B8rensen%E2%80%93Dice_coefficient.

Two additional methods are added to consider metric-based losses, however there are not used for now. The used loss functions are still binary cross-entropy for feature detection and categorical cross-entropy for semantic segmentation, as they are very popular in the state-of-the-art... Further investigations could be done in the future on this particular question, however we can consider that these loss functions are satisfying.

All the metrics are logger in a `.csv` file in instance-related repositories, so as to keep track of each training process.

This PR fixes issue #21 .

### CODE REVIEW

File: deeposlandia/train.py [hunk 1]
```diff
@@ -123,6 +123,96 @@ def add_training_arguments(parser):
...
+    Returns
+    -------
+    float
+        Intersection over Union value
+    """
+    actual = backend.flatten(actual)
+    predicted = backend.flatten(predicted)
+    intersection = backend.sum(actual * predicted)
+    union = backend.sum(actual) + backend.sum(predicted) - intersection
+    return 1. * intersection / union
```
Comment (garaud) [inline deeposlandia/train.py]: Can you tell my why you multiply by 1. ?

File: deeposlandia/train.py [hunk 2]
```diff
@@ -123,6 +123,96 @@ def add_training_arguments(parser):
...
+
+def iou_loss(actual, predicted):
```
Comment (garaud) [inline deeposlandia/train.py]: I think if we have more loss/metrics functions, we will move them into a dedicated Python module.

**Review by garaud — CHANGES_REQUESTED** [review]
Just a comment and one question. Looks good for me.

**Review by garaud — APPROVED** [review]

File: deeposlandia/train.py (hunk 1 above)
Comment (delhomer) [reply deeposlandia/train.py]: We want to get a float value, however `intersection` and `union` are tensors that contains integer values.

File: deeposlandia/train.py (hunk 2 above)
Comment (delhomer) [reply deeposlandia/train.py]: I agree. I have prepared a commit to put metric design methods in a `metrics.py` module.
